File: src/main/java/org/pl/lexer/tokenizer/MinusTokenizer.java
package org.pl.lexer.tokenizer;

import org.pl.lexer.ILexer;
import org.pl.lexer.exception.TokenizerException;
import org.pl.lexer.token.arithmetic.MinusToken;

/**
 * Tokenizes the minus symbol '-'.
 */
public class MinusTokenizer implements ITokenizer {
    @Override
    public boolean matches(Character c) {
        return c == '-';
    }

    @Override
    public MinusToken tokenize(ILexer lexer) throws TokenizerException {
        return new MinusToken();
    }
}

File: src/main/java/org/pl/lexer/tokenizer/MinusTokenizer.kt
package org.pl.lexer.tokenizer

import org.pl.lexer.ILexer
import org.pl.lexer.token.ArrowToken
import org.pl.lexer.token.IToken
import org.pl.lexer.token.arithmetic.MinusToken

/**
 * Tokenizes the minus symbol '-'.
 */
class MinusTokenizer : ITokenizer {
    override fun matches(c: Char) = c == '-'
    override fun tokenize(lexer: ILexer): IToken {
        if (lexer.peekNextSymbol == '>') {
            lexer.advanceCursor()
            return ArrowToken()
        }
        return MinusToken()
    }
}